src/Passes/Source/Rules/OperandPrototype.hpp

```diff
@@ -0,0 +1,105 @@
+#pragma once
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "Llvm/Llvm.hpp"
+
+#include <unordered_map>
+#include <vector>
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    /// IOperandPrototype describes an IR pattern and allows matching against
+    /// LLVMs llvm::Value type. Each value may or may not be captured during the
+    /// matching process which means that they are stored in a map under a given name.
+    /// Capturing is enabled using `enableCapture(name)` which sets the name the
+    /// value should be stored under.
+    class IOperandPrototype
```

Comment:
Consider naming identically the class and its files.

Reply:
Done.